cmd/zpool/zpool_main.c

```diff
@@ -2915,20 +3811,52 @@ zpool_do_iostat(int argc, char **argv)
 		(void) fprintf(stderr, gettext("no pools available\n"));
 		return (1);
 	}
-
 	/*
 	 * Enter the main iostat loop.
 	 */
 	cb.cb_list = list;
-	cb.cb_verbose = verbose;
-	if (guid)
-		cb.cb_name_flags |= VDEV_NAME_GUID;
-	if (follow_links)
-		cb.cb_name_flags |= VDEV_NAME_FOLLOW_LINKS;
-	if (full_name)
-		cb.cb_name_flags |= VDEV_NAME_PATH;
-	cb.cb_iteration = 0;
-	cb.cb_namewidth = 0;
+
+	/*
+	 * Histograms tables look out of place when you try to display them
+	 * with the other stats. Make a rule that you can only print
+	 * histograms by themselves.
+	 */
+	if (histo) {
+		cb.cb_flags = (histo << VS_HISTO);
+	} else {
+		cb.cb_flags = (1 << VS_DEFAULT) | (latency << VS_LATENCY) |
+				(queues << VS_QUEUES);
```

Comment:
The VS_LATENCY, VS_QUEUES and VS_HISTO macros have existing bitmask equivalents macros defined as VS_LATENCY_M, VS_QUEUES_M, and VS_HISTO_M.  I think I can use them to make this code easier to read.